django_netjsonconfig/base/admin.py

```diff
@@ -362,6 +365,62 @@ class AbstractTemplateAdmin(BaseConfigAdmin):
     class Media(BaseConfigAdmin.Media):
         js = BaseConfigAdmin.Media.js + [static('{0}js/{1}'.format(prefix, 'template_admin.js'))]
 
+    def delete_view(self, request, object_id, extra_context=None):
+        """
+        send unsubscription notifications to template designer
+        when deleted from the detail page.
+        """
+        if request.POST:
+            template = self.model.objects.get(pk=object_id)
+            subscriber_url = '{0}://{1}'.format(request.META.get('wsgi.url_scheme'),
+                                                request.get_host())
+            subscribe.delay(template.id, template.url, subscriber_url, subscribe=False)
+            self._delete_vpn(template.vpn_id)
+        return super(AbstractTemplateAdmin, self).delete_view(request, object_id, extra_context)
+
+    def subscribers(self, obj=None):
+        count = 0
+        if obj and obj.sharing == 'public' or obj.sharing == 'secret_key':
+            count = self.template_subscribe_model.objects.filter(subscribe=True, template=obj).count()
+        return count
+    subscribers.short_description = _('Number of Subscribers')
+
+    def save_model(self, request, obj, form, change):
+        """
+        Send notifications to template designers
+        for subscription.
+        """
+        super(AbstractTemplateAdmin, self).save_model(request, obj, form, change)
+        if not change and obj.sharing == 'import':
+            subscriber_url = '{0}://{1}'.format(request.META.get('wsgi.url_scheme'),
+                                                request.get_host())
+            subscribe.delay(obj.pk, obj.url, subscriber_url, subscribe=True)
+
+    def delete_selected(self, request, queryset):
+        """
+        send unsubscription notification to template designer
+        when template is deleted from the change list.
+        """
+        if request.POST.get('post'):
+            for template in queryset:
+                if template.sharing == 'import':
+                    subscriber_url = '{0}://{1}'.format(request.META.get('wsgi.url_scheme'),
+                                                        request.get_host())
+                    subscribe.delay(template.id, template.url, subscriber_url, subscribe=False)
+                    self._delete_vpn(template.vpn_id)
+        return delete_selected_(self, request, queryset)
+
+    delete_selected.short_description = _('Delete selected templates')
+
+    def _delete_vpn(self, vpn_id):
```

Comment:
I would move this to the `Template` model and call it `_delete_imported_vpn` and make sure it does not do anything if the VPN is associated to any other template which is not `self`